### PR TITLE
[Fix] Progressbars for encoding + hash encoding for low cardinality sets

### PR DESCRIPTION
**Pull request by Imipenem**
**PR Checklist**

-   [x] This comment contains a description of changes (with reason)
-   [x] Referenced issue is linked

**Description of changes**

- fixed an issue that caused progressbars to not update properly

- trying to encode low cardinality datasets with hash encoding now throws a custom error, explaining the failure instead of a traceback (background: hash encoding does not work well with such datasets)

### CODE REVIEW

File: ehrapy/api/preprocessing/encoding/_encode.py [hunk 1]
```diff
@@ -255,7 +253,7 @@ def _encode(
                             var_to_encoding[column_name] = encoding_mode
                     else:
                         var_to_encoding[categorical] = encoding_mode
-                    progress.update(task, advance=1)
+                # progress.advance(task, advance=1)
```
Comment (Imipenem) [inline ehrapy/api/preprocessing/encoding/_encode.py:256]:
```suggestion
  
```

File: ehrapy/api/preprocessing/encoding/_encode.py [hunk 2]
```diff
@@ -266,6 +264,7 @@ def _encode(
             categoricals,
         )
         try:
+            print(encoded_x)
```
Comment (Imipenem) [inline ehrapy/api/preprocessing/encoding/_encode.py:267]:
```suggestion
```

File: ehrapy/api/preprocessing/encoding/_encode.py [hunk 3]
```diff
@@ -228,13 +228,12 @@ def _encode(
...
         encoded_x = None
         encoded_var_names = adata.var_names.to_list()
-
         with Progress(
             "[progress.description]{task.description}", BarColumn(), "[progress.percentage]{task.percentage:>3.0f}%"
         ) as progress:
 
             for encoding_mode in encodings.keys():
-                task = progress.add_task(f"[red] Setting up {encoding_mode}", total=len(encodings[encoding_mode]))
+                task = progress.add_task(f"[red] Setting up {encoding_mode}", total=1)
```
Comment (Zethson) [inline ehrapy/api/preprocessing/encoding/_encode.py:236]: Is it possible to just remove the total here?
Isn't this like total=1?

Comment (Imipenem) [reply ehrapy/api/preprocessing/encoding/_encode.py:236]: From rich docs: `add_task(description: str, start: bool = True, total: float = 100.0, completed: int = 0, visible: bool = True, **fields: Any) → TaskID`.

Defaults to `100.0`

**Pull request merged**